config/src/iroha.rs

```diff
@@ -151,79 +148,101 @@ impl ConfigurationProxy {
+
+    /// Key-pair used for proptests generation
+    #[allow(clippy::expect_used)]
+    pub fn placeholder_keypair() -> KeyPair {
+        let public_key = "ed01201c61faf8fe94e253b93114240394f79a607b7fa55f9e5a41ebec74b88055768b"
+            .parse()
+            .expect("Public key not in mulithash format");
+        let private_key = PrivateKey::from_hex(
+            Algorithm::Ed25519,
+            "282ed9f3cf92811c3818dbc4ae594ed59dc1a2f78e4241e31924e101d6b1fb831c61faf8fe94e253b93114240394f79a607b7fa55f9e5a41ebec74b88055768b"
+        ).expect("Private key not hex encoded");
+
+        KeyPair::new(public_key, private_key).expect("Key pair mismatch")
+    }
+
+    fn arb_keys() -> BoxedStrategy<(Option<PublicKey>, Option<PrivateKey>)> {
+        let (pub_key, priv_key) = placeholder_keypair().into();
+        (
+            prop::option::of(Just(pub_key)),
+            prop::option::of(Just(priv_key)),
+        )
+            .boxed()
+    }
+
+    prop_compose! {
+        fn arb_proxy()(
+            (public_key, private_key) in arb_keys(),
+            disable_panic_terminal_colors in prop::option::of(Just(true)),
+            kura in prop::option::of(kura::tests::arb_proxy()),
+            sumeragi in prop::option::of(sumeragi::tests::arb_proxy()),
+            torii in prop::option::of(torii::tests::arb_proxy()),
+            block_sync in prop::option::of(block_sync::tests::arb_proxy()),
+            queue in prop::option::of(queue::tests::arb_proxy()),
+            logger in prop::option::of(logger::tests::arb_proxy()),
+            genesis in prop::option::of(genesis::tests::arb_proxy()),
+            wsv in prop::option::of(wsv::tests::arb_proxy()),
+            network in prop::option::of(network::tests::arb_proxy()),
+            telemetry in prop::option::of(telemetry::tests::arb_proxy()),
+            ) -> ConfigurationProxy {
+            ConfigurationProxy { public_key, private_key, disable_panic_terminal_colors, kura, sumeragi, torii, block_sync, queue,
+                                 logger, genesis, wsv, network, telemetry }
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn iroha_proxy_build_fails_on_none(proxy in arb_proxy()) {
+            let cfg = proxy.build();
+            let example_cfg = ConfigurationProxy::from_path(CONFIGURATION_PATH)
+                .expect("Failed to read example config file").build().expect("Failed to build example Iroha config");
+            if cfg.is_ok() {
+                assert_eq!(cfg.unwrap(), example_cfg)
+            }
+        }
+    }
 
     #[test]
     fn parse_example_json() -> Result<()> {
         let cfg_proxy = ConfigurationProxy::from_path(CONFIGURATION_PATH)
             .wrap_err("Failed to read configuration from example config")?;
-        assert_eq!(
-            "./storage",
-            cfg_proxy.kura.unwrap().block_store_path.unwrap()
-        );
+        assert_eq!("127.0.0.1:1337", cfg_proxy.torii.unwrap().p2p_addr.unwrap());
```

Comment:
Why do we switch this line back and forth?

Reply:
Formatting. Small changes lead to it being either written on separate lines, or on the same line.

Reply:
Actually it's not about formatting here, but due to me mindlessly rewriting back changes introduced by [this](https://github.com/hyperledger/iroha/pull/2782#discussion_r980902572). Will revert it back.